lumicks/pylake/tests/test_imaging_mixins.py

```diff
@@ -0,0 +1,209 @@
+import pytest
+import numpy as np
+
+from lumicks.pylake.detail.imaging_mixins import TiffExport
+
+
+def tiffexport_factory(
+    images,
+    frames=True,
+    image_metadata=True,
+    timestamp_ranges=True,
+    writer_kwargs=True,
+    start_time=0,
+    exposure_time=5,
+    frame_time=10,
+):
+    def tiff_frames_factory(frames):
+        def tiff_frames(iterator=False):
+            if not iterator:
+                return frames
+
+            def frames_iterator():
+                for frame in frames:
+                    yield frame
+
+            return frames_iterator()
+
+        return tiff_frames
+
+    def tiff_image_metadata():
+        return {"Writing": "tests is awesome!"}
+
+    def tiff_timestamp_ranges_factory(number_of_frames):
+        def tiff_timestamp_ranges():
+            return [
+                np.array([i, i + exposure_time])
+                for i in range(
+                    start_time, start_time + frame_time * number_of_frames, start_time + frame_time
+                )
+            ]
+
+        return tiff_timestamp_ranges
```

Comment:
normally I might argue against a multiline `lambda` but to save on the nesting might be nice here
```suggestion
    def tiff_timestamp_ranges_factory(number_of_frames):
        return lambda: [
            np.array([i, i + exposure_time])
            for i in range(
                start_time, start_time + frame_time * number_of_frames, start_time + frame_time
            )
        ]
```

Reply:
I do not like lambdas and am not so sure if the lambda improves the nesting situation. I like the verbose definition of the function `tiff_timestamp_ranges` that is finally returned. Should we vote? @JoepVanlier

Reply:
My vote for this one would go to no lambda.

Reply:
ok yeah here I'm fine keeping the nested function